docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,246 @@
+  - Type `#autograder` to connect the issue to the Autograder
+  - Type a title and hit enter
+- Use a clear, descriptive title that uses one of these prefixes:
+  - `Backend:` server side changes
+  - `Frontend:` client (web) side changes
+  - `Fullstack:` for issues that require front and backend changes
+  - `Docs:` changes to documentation (either markdown files or JSDocs/JavaDocs)
+  - `Tests:` adding or modifying tests
+  - `Dev:` changes to enhance the Autograder development experience
+- Provide enough context so others can understand the purpose
+- Add labels as appropriate ([Here's the list](https://github.com/softwareconstruction240/autograder/issues/labels))
+- Link related issues
+- Issues should be narrow in scope
+
+#### Working on an Issue
+- Assign the issue to yourself before beginning work (this is how we know 
+  who is working on what, and gives you a sort of ownership over the issue)
+- Create a new branch ([more info here](#2-branches))
+- If you find a new issue, file it separately (don't mix concerns)
+- Leave a comment from time to time with updates on your progress
+
+### 2. Branches
+Branches in git allow us to all work on separate things in the same codebase.
+You will do all your work in a branch.
+
+#### Creating a branch
+Create the branch directly from the issue so your work is linked.
+
+- Always branch from `main` 
+  - (exceptions exist if you know what you're doing)
+- Use descriptive names that reflect what you're working on:
+    - `add-late-submission`
+    - `admin-display-verification-status`
+    - `server-communicator`
+    - `extract-service-logic-from-controllers`
+- Use kebab-case (lowercase with hyphens)
+- Keep names concise but clear
+- One branch per issue (don't mix different features/fixes)
+
+### 3. Making Changes
+Now that you have a branch, you can start to code! Hurray! 🎉
+Here are important guidelines to follow:
+
+- Review and follow the Style Guide (_Coming Soon_)
+- Write tests for your code. All new features and functionality must have tests demonstrating correctness
+  - Update existing tests if you change functionality
+  - _New contributors: A great way to better understand the codebase is to try writing tests for existing code_
+- Keep commits focused and meaningful
+  - Each commit should represent one logical change
+  - Write clear commit messages that describe that change ([See guidelines](#writing-good-commit-messages))
+- Update documentation as you go
+  - Add JSDoc/JavaDoc comments to new code
+    - _New contributors: Another great way to learn the codebase is to document existing code_
+  - Update repo Markdown docs if needed
+  - Code should for the most part be self-documenting. If not, make sure to add inline comments
+- Push your code to GitHub at the end of each coding session. This lets other
+  team members see how you're doing
+
+#### Writing Good Commit Messages
+- Start with a present-tense verb that describes what the commit does:
```

Comment:
Good point, @ThanGerlek. I had to ask to discover the difference, and I was able to verify that "imperative statement" is the standard.

## In the context of authoring git commit messages, what's the difference between using "a present tense verb" and an "imperative statement?"

> Credit: The following research was kindly generated by [ChatGPT](https://chatgpt.com/share/672c7559-df04-800c-8ea3-60a17daad887).

In git commit messages, using a *present tense verb* and an *imperative statement* often overlap, but they serve slightly different purposes:

1. **Present Tense Verb**: Using the present tense describes what the commit does. For example, saying "Fixes bug in login flow" states in a straightforward way what the commit achieves.

2. **Imperative Statement**: Using the imperative mood (e.g., "Fix bug in login flow") gives a command or instruction, as if you’re telling Git (or a reader) what to do. This is preferred in git commit messages because it reads like an instruction—"If applied, this commit will..."—and maintains consistency with Git’s own conventions and tooling.

The imperative mood usually implies present tense, so in practice, it’s both present and imperative, making it a natural choice for clarity.

Reply:
fixed

Reply:
I knew the way I had it written was weird, thanks for pointing this out. 

fixed